clients/src/main/java/org/apache/kafka/clients/producer/internals/RecordAccumulator.java

```diff
@@ -272,37 +281,47 @@ private boolean isMuted(TopicPartition tp, long now) {
         return result;
     }
 
+    private void maybeUpdateNextBatchExpiryTime(ProducerBatch batch) {
+        nextBatchExpiryTimeMs = (batch.createdMs + deliveryTimeoutMs < 0) ? nextBatchExpiryTimeMs
```

Comment:
In what scenario will `batch.createdMs + deliveryTimeoutMs` be negative? `batch.createdMs` is the wall clock time when the batch is created, and should always be positive. The config def for `deliveryTimeoutMs` enforces the value is `atleast(0)`. So then how could the sum of the two be negative? Are you accounting for wrap around?

Reply:
Ok. It would be good mention this in a comment. It is very non obvious.

Reply:
update the code to use `if ...` statement and add the comment